staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -74,6 +74,8 @@ type UpgradeAwareHandler struct {
 	RequireSameHostRedirects bool
 	// UseRequestLocation will use the incoming request URL when talking to the backend server.
 	UseRequestLocation bool
+	// OverrideRequestHost will override the HTTP host header when talking to the backend server.
+	OverrideRequestHost bool
```

Comment:
```suggestion
	// UseLocationHost overrides the HTTP host header in requests to the backend server to use the Host from Location
	UseLocationHost bool
```

Reply:
also document how this interacts with the `UseRequestLocation` option (it's not clear to me how it should)

Reply:
@liggitt I changed it like you suggested and added the explanation from the [net/http/request.go](https://github.com/golang/go/blob/d4b26382342c98a95b85140b2863bc30c48edd68/src/net/http/request.go#L119-L122) comment about the difference between req.URL.Host (set by `UseRequestLocation`) and req.Host (set by `UseLocationHost`)